models/campaign.go

```diff
@@ -471,6 +471,44 @@ func DeleteCampaign(id int64) error {
 	return err
 }
 
+//DeleteCampaignByUserId deletes campaigns with a given user id
+func DeleteCampaignByUserId(uid int64) error {
+	// Get campaigns
+	cs, err := GetCampaigns(uid)
+	if err != nil {
+		Logger.Println(err)
+		return err
+	}
+
+	var cids []int64
+	for _, c := range cs {
+		cids = append(cids, c.Id)
+	}
+
+	// Delete all the results
+	err = db.Where("campaign_id in (?)", cids).Delete(&Result{}).Error
```

Comment:
Hmmm... I wish you didn't have to delete results this way. I'm wondering if it'd be better for me to create a SQL migration that adds a cascading delete condition to the results -> campaign relationship. What do you think?